libsolutil/InvertibleMap.h

```diff
@@ -32,26 +32,35 @@ struct InvertibleMap
 
 	void set(K _key, V _value)
 	{
-		if (values.count(_key))
-			references[values[_key]].erase(_key);
-		values[_key] = _value;
+		auto it = values.find(_key);
+		if (it == values.end())
+			values[_key] = _value;
```

Comment:
We could replace `find` by `upper_bound` and a comparison, then the insertion here can use the hint.